### PR TITLE
Update publish.yaml and remove deprecated field in metadata.yaml (cannon/change-50b69e88)

### PR DESCRIPTION
**Pull request by ca-scribner**
Changes applied by commit-cannon:
  * Ran command `find . -name metadata.yaml -exec sed -i '/display-name/d' {} \;`
  * Replaced file `.github/workflows/publish.yaml`

### CODE REVIEW

File: .github/workflows/publish.yaml [hunk 1]
```diff
@@ -4,33 +4,25 @@ on:
...
+      - track/**
 
 jobs:
-  charm:
+  publish-charm:
     name: Publish Charm
     runs-on: ubuntu-latest
-
+    # Only publish to charmhub if we are pushing to a special branch or running PRs from something named `branch/*`
+    if: (github.event_name == 'push') ||  (startsWith( github.head_ref, 'branch/' ))
```
Comment (DnPlas) [inline .github/workflows/publish.yaml:20]: Can this check be omitted by adding `branch/` to the `push` block above?

**Review by DnPlas — COMMENTED** [review]
Just a comment, thanks @ca-scribner

File: .github/workflows/publish.yaml (hunk 1 above)
Comment (ca-scribner) [reply .github/workflows/publish.yaml:20]: No I don't think so :( 

The entries in `on` above are applied with `or` (`condition A or condition B or ...`) but we need an `and` (`PR against main/master/track and PR is from branch named branch/*`)

Comment (DnPlas) [reply .github/workflows/publish.yaml:20]: Alright, makes sense. Thanks

**Comment by DnPlas** [comment]
LGTM, thanks @ca-scribner

**Pull request merged**